src/components/Search.tsx

```diff
@@ -5,10 +5,8 @@ type SearchProps = { initialQuery: string; onSearch: (query: string) => void }
 export function Search({ initialQuery, onSearch }: SearchProps) {
     function handleSearch(event: FormEvent<HTMLFormElement>) {
         event.preventDefault()
-        const target = event.target as typeof event.target & {
-            searchInput: { value: string }
-        }
-        onSearch(target.searchInput.value)
+        const data = new FormData(event.currentTarget)
+        onSearch(data.get('searchInput') as string)
```

Comment:
I'm not sure if its the best solution in terms of type safety. If someday we have many elements, we will need to cast a lot with this approach. 
Check out how Ken Dodds handles this: https://epicreact.dev/how-to-type-a-react-form-on-submit-handler/